File: uplift-gradle-plugins/src/main/java/com/github/kjetilv/uplift/plugins/UpliftInitTask.java
package com.github.kjetilv.uplift.plugins;

import org.gradle.api.provider.Property;
import org.gradle.api.tasks.CacheableTask;
import org.gradle.api.tasks.OutputFile;

import java.nio.file.Path;

/** Generates the CDK application. Touches no AWS resources. */
@CacheableTask
public abstract class UpliftInitTask extends UpliftCdkTask {

    @OutputFile
    public abstract Property<Path> getActiveJar();

    @OutputFile
    public abstract Property<Path> getActivePom();

    @Override
    protected void perform() {
        initCdkApp();
    }
}

File: uplift-gradle-plugins/src/main/kotlin/com/github/kjetilv/uplift/plugins/UpliftInitTask.kt
package com.github.kjetilv.uplift.plugins

import org.gradle.api.provider.Property
import org.gradle.api.tasks.CacheableTask
import org.gradle.api.tasks.OutputFile
import org.gradle.process.ExecOperations
import java.nio.file.Path
import javax.inject.Inject

@CacheableTask
abstract class UpliftInitTask @Inject constructor(execOperations: ExecOperations) : UpliftCdkTask(execOperations) {

    @get:OutputFile
    abstract val activeJar: Property<Path>

    @get:OutputFile
    abstract val activePom: Property<Path>

    override fun perform() = initCdkApp()
}
